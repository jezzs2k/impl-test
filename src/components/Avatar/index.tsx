import React from 'react';
import {Image, TouchableOpacity, ViewStyle, StyleProp} from 'react-native';

import {widthPercentageToDP} from '../../utils';

export const LINK_IMAGE_DEFAULT =
  'https://interactive-examples.mdn.mozilla.net/media/cc0-images/grapefruit-slice-332-332.jpg';

type AvatarPropType = {
  uri?: string;
  size?: number;
  styleAvatar?: StyleProp<ViewStyle>;
  borderWidth?: number;
  onPress?: () => void;
  activeOpacity?: number;
  borderColor?: string;
};

export const Avatar = ({
  uri,
  size,
  styleAvatar,
  borderWidth,
  onPress,
  activeOpacity,
  borderColor,
}: AvatarPropType) => (
  <TouchableOpacity
    style={[styleAvatar]}
    onPress={onPress}
    accessibilityRole={'button'}
    activeOpacity={activeOpacity}>
    <Image
      source={{uri}}
      style={{
        width: size,
        height: size,
        borderRadius: size / 2,
        borderWidth: borderWidth,
        borderColor: borderColor,
      }}
    />
  </TouchableOpacity>
);

Avatar.defaultProps = {
  uri: LINK_IMAGE_DEFAULT,
  size: widthPercentageToDP(20),
  styleAvatar: {},
  borderWidth: 0,
  onPress: () => {},
  activeOpacity: 1,
  borderColor: '#fff',
};
