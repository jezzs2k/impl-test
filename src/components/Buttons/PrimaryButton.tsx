import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Text,
  ActivityIndicator,
  StyleProp,
  ViewStyle,
  TextStyle,
} from 'react-native';

import {heightPercentageToDP, responsiveFontWidth} from '../../utils';

export interface PrimaryButtonPropType {
  titleBtn: string;
  titleStyle?: StyleProp<ViewStyle | TextStyle>;
  onPress: () => void;
  disabled?: boolean;
  btnStyle?: StyleProp<ViewStyle>;
  loading?: boolean;
}

export const PrimaryButton = ({
  titleBtn,
  titleStyle = {},
  onPress,
  disabled = false,
  btnStyle = {},
  loading = false,
}: PrimaryButtonPropType) => {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      disabled={disabled}
      onPress={onPress}
      style={[styles.container, btnStyle, disabled && styles.disableStyle]}>
      {loading ? (
        <ActivityIndicator size={'small'} />
      ) : (
        <Text
          style={[
            styles.titleStyle,
            titleStyle,
            disabled && styles.disableText,
          ]}>
          {titleBtn}
        </Text>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 50,
    paddingHorizontal: 25,
    backgroundColor: '#000',
  },
  titleStyle: {
    color: 'white',
    fontSize: responsiveFontWidth(4.8),
    paddingVertical: heightPercentageToDP(1),
  },
  disableStyle: {
    backgroundColor: '#aaa69d',
  },
  disableText: {
    color: '#d1ccc0',
  },
});
