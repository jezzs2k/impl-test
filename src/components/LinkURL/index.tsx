import React, {useCallback} from 'react';
import {
  Linking,
  StyleProp,
  Text,
  TouchableOpacity,
  TextStyle,
  StyleSheet,
} from 'react-native';
import {responsiveFontSetting} from '../../utils';

type LinkURLProptype = {
  url: string;
  styleUrl?: StyleProp<TextStyle>;
  title?: string;
};

export const LinkURL = ({
  url,
  styleUrl = {},
  title = 'Brower',
}: LinkURLProptype) => {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url);

    if (supported) {
      await Linking.openURL(url);
    }
  }, [url]);

  return (
    <TouchableOpacity onPress={handlePress}>
      <Text style={[styles.textSmall, styles.textBold, styles.link, styleUrl]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
  },
  link: {
    fontStyle: 'italic',
    textTransform: 'capitalize',
    fontSize: responsiveFontSetting(2),
    color: '#1068bf',
    textDecorationLine: 'underline',
  },
  textSmall: {
    fontSize: responsiveFontSetting(1.6),
    fontStyle: 'normal',
  },
});
