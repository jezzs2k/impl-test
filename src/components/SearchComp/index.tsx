import React from 'react';
import {View, TextInput, Text, StyleSheet} from 'react-native';
import {
  heightPercentageToDP,
  responsiveFontSetting,
  widthPercentageToDP,
} from '../../utils';

type SearchCompPropType = {
  onchangeText: (text: string) => void;
  textTitle: string;
  placeholder?: string;
};

export const SearchComp = ({
  onchangeText,
  textTitle,
  placeholder,
}: SearchCompPropType) => {
  return (
    <View style={styles.container}>
      <View style={styles.containerItem}>
        <Text style={styles.textBold}>{textTitle}</Text>
      </View>
      <TextInput
        style={styles.textInputStyle}
        placeholder={placeholder ?? 'Search user ...'}
        onChangeText={onchangeText}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    paddingHorizontal: widthPercentageToDP(5),
    paddingVertical: heightPercentageToDP(1),
  },
  containerItem: {
    marginBottom: heightPercentageToDP(1),
    paddingTop: heightPercentageToDP(1),
  },
  textStyle: {
    fontSize: responsiveFontSetting(2),
  },
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
    textTransform: 'capitalize',
  },
  textInputStyle: {
    paddingVertical: heightPercentageToDP(1),
    paddingHorizontal: widthPercentageToDP(3),
    borderWidth: 1,
    fontSize: responsiveFontSetting(2),
    borderRadius: 8,
    borderColor: '#bdc3c7',
  },
});
