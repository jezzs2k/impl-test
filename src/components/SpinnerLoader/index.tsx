import React from 'react';
import {View, StyleSheet} from 'react-native';
import Svg, {Rect} from 'react-native-svg';
import ContentLoader from 'react-native-masked-loader';

const getMaskedElement = () => {
  return (
    <Svg height={'250px'} width="100%" fill={'black'}>
      <Rect x="0" y="0" rx="8" ry="8" width="50%" height="16" />
      <Rect x="0" y="30" rx="9" ry="9" width="100%" height="128" />
      <Rect x="0" y="172" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="188" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="204" rx="4" ry="4" width="100%" height="8" />
    </Svg>
  );
};

export const BOSLoader = () => {
  const MaskedElement = getMaskedElement();
  return (
    <View style={styles.container}>
      <ContentLoader MaskedElement={MaskedElement} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    height: 285,
  },
});
