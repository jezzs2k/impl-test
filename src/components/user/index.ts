export * from './userDetailHeader';
export * from './userItem';
