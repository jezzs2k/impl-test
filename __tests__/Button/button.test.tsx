import React from 'react';
import renrender from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import {PrimaryButton} from '../../src/components/Buttons';

describe('Buttons', () => {
  it('Snapshot test button component', () => {
    const mockFn = jest.fn();
    const component = renrender.create(
      <PrimaryButton titleBtn={'Button'} onPress={mockFn} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Click Button', () => {
    const mockFn = jest.fn();

    const {getByText} = render(
      <PrimaryButton titleBtn={'Button'} onPress={mockFn} />,
    );

    fireEvent.press(getByText('Button'));

    expect(mockFn).toHaveBeenCalled();
  });
  it('Disable click Button', () => {
    const mockFn = jest.fn();

    const {getByText} = render(
      <PrimaryButton titleBtn={'Button'} onPress={mockFn} disabled />,
    );

    fireEvent.press(getByText('Button'));

    expect(mockFn).toHaveBeenCalledTimes(0);
  });
});
