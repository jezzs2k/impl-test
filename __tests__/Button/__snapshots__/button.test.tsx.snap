// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Snapshot test button component 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 25,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 18,
          "paddingVertical": 13.5,
        },
        Object {},
        false,
      ]
    }
  >
    Button
  </Text>
</View>
`;
